include/evaluators/blas_functor_traits.hpp

```diff
@@ -0,0 +1,144 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas_functor_traits.hpp
+ *
+ **************************************************************************/
+
+#ifndef BLAS_FUNCTOR_TRAITS_HPP_7FBGTUQ8
+#define BLAS_FUNCTOR_TRAITS_HPP_7FBGTUQ8
+
+#include <iostream>
+#include <stdexcept>
+#include <vector>
+
+#include <CL/sycl.hpp>
+
+#include <executors/blas_packet_traits_sycl.hpp>
+#include <operations/blas_operators.hpp>
+
+namespace blas {
+
+template <class Functor, typename ScalarT, typename Device>
+struct functor_traits : Functor {
+  static constexpr bool supported = false;
+};
+
+template <typename ScalarT, typename Device>
+struct functor_traits<iniAddOp1_struct, ScalarT, Device> : iniAddOp1_struct {
+  static constexpr bool supported = true;
+};
+template <typename ScalarT, typename Device>
+struct functor_traits<iniPrdOp1_struct, ScalarT, Device> : iniPrdOp1_struct {
+  static constexpr bool supported = true;
+};
+template <typename ScalarT, typename Device>
+struct functor_traits<posOp1_struct, ScalarT, Device> : posOp1_struct {
+  static constexpr bool supported = true;
+};
+template <typename ScalarT, typename Device>
+struct functor_traits<negOp1_struct, ScalarT, Device> : negOp1_struct {
+  static constexpr bool supported = Packet_traits<ScalarT, Device>::has_neg;
+};
+template <typename ScalarT, typename Device>
+struct functor_traits<sqtOp1_struct, ScalarT, Device> : sqtOp1_struct {
+  static constexpr bool supported = Packet_traits<ScalarT, Device>::has_sqrt;
+  template <typename R = ScalarT>
+  static R eval(R r) {
+    return cl::sycl::sqrt(r);
+  }
+};
+template <typename ScalarT, typename Device>
+struct functor_traits<tupOp1_struct, ScalarT, Device> : tupOp1_struct {
+  static constexpr bool supported = true;
+};
+template <typename ScalarT, typename Device>
+struct functor_traits<addOp1_struct, ScalarT, Device> : addOp1_struct {
+  static constexpr bool supported = Packet_traits<ScalarT, Device>::has_add ||
+                                    Packet_traits<ScalarT, Device>::has_mul;
+};
+template <typename ScalarT, typename Device>
+struct functor_traits<prdOp1_struct, ScalarT, Device> : prdOp1_struct {
+  static constexpr bool supported = Packet_traits<ScalarT, Device>::has_mul;
+};
+template <typename ScalarT, typename Device>
+struct functor_traits<absOp1_struct, ScalarT, Device> : absOp1_struct {
+  static constexpr bool supported = Packet_traits<ScalarT, Device>::has_abs;
+  template <typename R = ScalarT>
+  static R eval(R r) {
+    return cl::sycl::abs(r);
+  }
+};
+template <typename ScalarT, typename Device>
+struct functor_traits<addOp2_struct, ScalarT, Device> : addOp2_struct {
+  static constexpr bool supported = Packet_traits<ScalarT, Device>::has_add;
```

Comment:
Packet_traits is a new thing on this merge request too? Can it be done on a separate one? 
Also Packet_traits is an inconsistent naming scheme, use either packet_traits or PacketTraits (I believe the first one is better)